src/ParseLiveQuery.js

```diff
@@ -98,13 +99,9 @@ const DefaultLiveQueryController = {
 
     // If we can not find Parse.liveQueryServerURL, we try to extract it from Parse.serverURL
     if (!liveQueryServerURL) {
-      const tempServerURL = CoreManager.get('SERVER_URL');
-      let protocol = 'ws://';
-      if (tempServerURL.indexOf('https') === 0) {
-        protocol = 'wss://'
-      }
-      const host = tempServerURL.replace(/^https?:\/\//, '');
-      liveQueryServerURL = protocol + host;
+      const serverURL = url.parse(CoreManager.get('SERVER_URL'));
+      const protocol = serverURL.protocol === 'http:' ? 'ws://' : 'wss://';
+      liveQueryServerURL = protocol + serverURL.host + serverURL.pathname;
```

Comment:
hot!

Reply:
uh @dplewis...wait a sec...is this on the client?  will the `url` class be available?

I'm so used to all the parse stuff being on the server, I get a little confused on what's where when it comes to the JS SDK...😬

Reply:
I forgot about that too. I keep forgetting about other environments besides node.

To answer your question I don't think so. I'll revert it in a PR.

Reply:
or https://www.npmjs.com/package/url-polyfill

the code is sooooo much better this way.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/URL

not sure what compatibility we're looking for.